arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -0,0 +1,770 @@
+
+  ret_string = rebuildPath(suffix);
+
+  return ret_string;
+
+} // RocksDBHotBackup::buildDirectoryPath
+
+
+std::string RocksDBHotBackup::rebuildPath(const std::string & suffix) {
+  std::string ret_string;
+
+  ret_string = getDatabasePath();
+  ret_string += TRI_DIR_SEPARATOR_CHAR;
+  ret_string += "hotbackups";
+
+  // This prefix path must exist, ignore errors
+  long sysError;
+  std::string errorStr;
+  TRI_CreateRecursiveDirectory(ret_string.c_str(), sysError, errorStr);
+
+  ret_string += TRI_DIR_SEPARATOR_CHAR;
+  ret_string += suffix;
+
+  return ret_string;
+
+} // RocksDBHotBackup::rebuildPath
+
+
+//
+// @brief Remove file or dir currently occupying "path"
+//
+bool RocksDBHotBackup::clearPath(const std::string & path) {
+  bool retFlag = {true};
+
+  if (basics::FileUtils::exists(path)) {
+    if (basics::FileUtils::isDirectory(path)) {
+      TRI_RemoveDirectory(path.c_str());
+    } else {
+      basics::FileUtils::remove(path);
+    } // else
+
+    // test if still there and error out?
+    if (basics::FileUtils::exists(path)) {
+      retFlag = false;
+      LOG_TOPIC(ERR, arangodb::Logger::ENGINES)
+        << "RocksDBHotBackup::clearPath:  unable to remove previous " << path;
+    } // if
+  } // if
+
+  return retFlag;
+
+} // RocksDBHotBackup::clearPath
+
+
+//
+// @brief Common routine for retrieving parameter from request body.
+//        Assumes caller has maintain state of _body and _valid
+//
+void RocksDBHotBackup::getParamValue(const char * key, std::string & value, bool required) {
+  VPackSlice tempSlice;
+
+  try {
+    if (_body.isObject() && _body.hasKey(key)) {
+      tempSlice = _body.get(key);
+      value = tempSlice.copyString();
+    } else if (required) {
+      if (_valid) {
+        _result.add(VPackValue(VPackValueType::Object));
+        _valid = false;
+      }
+      _result.add(key, VPackValue("parameter required"));
+    } // else if
+  } catch(VPackException const &vexcept) {
+    if (_valid) {
+      _result.add(VPackValue(VPackValueType::Object));
+      _valid = false;
+    }
+    _result.add(key, VPackValue(vexcept.what()));
+  };
+
+} // RocksDBHotBackup::getParamValue (std::string)
+
+
+void RocksDBHotBackup::getParamValue(const char * key, bool & value, bool required) {
+  VPackSlice tempSlice;
+
+  try {
+    if (_body.isObject() && _body.hasKey(key)) {
+      tempSlice = _body.get(key);
+      value = tempSlice.getBool();
+    } else if (required) {
+      if (_valid) {
+        _result.add(VPackValue(VPackValueType::Object));
+        _valid = false;
+      }
+      _result.add(key, VPackValue("parameter required"));
+    } // else if
+  } catch(VPackException const &vexcept) {
+    if (_valid) {
+      _result.add(VPackValue(VPackValueType::Object));
+      _valid = false;
+    }
+    _result.add(key, VPackValue(vexcept.what()));
+  };
+
+} // RocksDBHotBackup::getParamValue (bool)
+
+
+void RocksDBHotBackup::getParamValue(const char * key, unsigned & value, bool required) {
```

Comment:
Just a comment, no change needed: I usually refrain from using "unsigned int" and prefer to always use the types which explicitly specify the bit width. It is much easier to reason about this to judge if overflow could happen. Here it is not a problem. So no change needed.